apps/storefront/src/pages/MyOrders/index.mobile.test.tsx

```diff
@@ -408,8 +408,9 @@ describe('when a personal customer', () => {
         expect(screen.getByRole('heading', { name: /66996/ })).toBeInTheDocument();
       });
 
-      expect(getOrders).toHaveBeenCalledTimes(3);
-      expect(getOrders).toHaveBeenLastCalledWith(expect.stringContaining('search: "66996"'));
+      await waitFor(() => {
+        expect(getOrders).toHaveBeenLastCalledWith(expect.stringContaining('search: "66996"'));
+      });
```

Comment:
something still odd to me here

```tsx
      await waitFor(() => {
        expect(getOrders).toHaveBeenLastCalledWith(expect.stringContaining('search: "66996"'));
      });
      console.log(getOrders.mock.calls.length);
```

logs `3` when ran with `it.only`
logs `4` when ran normally

🤔

Reply:
After further investigation, all these issues are related to a debounced function in the `B3FilterSearch` component. Moving to using `vitest-when` resolves the issue, as it will only return the right value if the query match the expected one.

The sequence of events that was causing this issue was:

1. Page mounts - fetch orders
2. refetch 🤷 
3. Mock the response
4. Find the textarea
5. Type in the textarea
6. remount - refetch -> gets the expected response
7. check the response value -> test fails
8. 500ms debounce finishes, the expected request happens